### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by johnmcollier**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+#### Repository Format
+
+The `registry-support` repository for the devfile registry is a mono-repo of sorts and there are a number of individual components contained within.
+
+- [Devfile Registry Build Tools](./build-tools) - Tools for building devfile registries into container images.
+- [Devfile Registry Helm Chart](./deploy/chart/devfile-registry) - Helm chart for deploying the devfile registry on to Kubernetes.
+- [Index Generator](./index/generator) - The index generator tool for generation index.json files from registry data
+- [Index Server](./index/server) - The index server, one of two servers that the devfile registry runs
+- [OCI Registry Server](./oci-registry) -
```
Comment (michael-valdron) [inline CONTRIBUTING.md:27]: Missing the description for the OCI Registry Server.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+To run these tests, consult the integration test's [readme](./tests/integration).
+
+### Submitting Pull Request
+
+**Note:** All commits must be signed off with the footer:
+```
+Signed-off-by: First Lastname <email@email.com>
+```
+
+You can easily add this footer to your commits by adding `-s` when running `git commit`.When you think the code is ready for review, create a pull request and link the issue associated with it.
```
Comment (michael-valdron) [inline CONTRIBUTING.md:59]: Just missing a space between the sentences.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+Owners of the repository will watch out for and review new PRs. 
+
+By default for each change in the PR, GitHub Actions and OpenShift CI will run checks against your changes (linting, unit testing, and integration tests)
```
Comment (michael-valdron) [inline CONTRIBUTING.md:63]: Nitpick: Just missing a period at the end of this sentence.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+If comments have been given in a review, they have to be addressed before merging.
+
+After addressing review comments, don’t forget to add a comment in the PR afterward, so everyone gets notified by Github and know to re-review.
+
+
+# Contact us
+
+If you have questions, please visit us on `#devfile` on the [Kubernetes Slack](https://slack.k8s.io).
```
Comment (michael-valdron) [inline CONTRIBUTING.md:72]: Nitpick: Would replace one of the "on"s with "in".

File: CONTRIBUTING.md (hunk 3 above)
Comment (michael-valdron) [inline CONTRIBUTING.md:63]: Nitpick: Just missing a period at the end of this sentence.

File: CONTRIBUTING.md (hunk 2 above)
Comment (michael-valdron) [inline CONTRIBUTING.md:59]: Just missing a space between the sentences.

File: CONTRIBUTING.md (hunk 1 above)
Comment (michael-valdron) [inline CONTRIBUTING.md:27]: Missing the description for the OCI Registry Server.

**Review by michael-valdron — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 3 above)
Comment (johnmcollier) [reply CONTRIBUTING.md]: Fixed.

File: CONTRIBUTING.md (hunk 4 above)
Comment (johnmcollier) [reply CONTRIBUTING.md]: Fixed.

File: CONTRIBUTING.md (hunk 2 above)
Comment (johnmcollier) [reply CONTRIBUTING.md]: Fixed.

File: CONTRIBUTING.md (hunk 1 above)
Comment (johnmcollier) [reply CONTRIBUTING.md]: Fixed.

**Review by michael-valdron — APPROVED** [review]
/lgtm

**Pull request merged**